File: react-native-pytorch-core/android/src/main/java/org/pytorch/rn/core/image/IImage.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package org.pytorch.rn.core.image;

import android.graphics.Bitmap;
import android.media.Image;
import androidx.annotation.Nullable;

public interface IImage extends AutoCloseable {
  float getPixelDensity();

  int getWidth();

  int getHeight();

  float getNaturalWidth();

  float getNaturalHeight();

  IImage scale(float sx, float sy);

  Bitmap getBitmap();

  @Nullable
  Image getImage();

  int getImageRotationDegrees();
}

File: react-native-pytorch-core/ios/Image/IImage.swift
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import Foundation

public protocol IImage {
    func getPixelDensity() -> CGFloat
    func getWidth() -> CGFloat
    func getHeight() -> CGFloat
    func getNaturalWidth() -> Int
    func getNaturalHeight() -> Int
    func scale(sx: CGFloat, sy: CGFloat) throws -> IImage
    func getBitmap() -> CGImage?
    func close() throws
}
